src/software/utils/main_colorCheckerDetection.cpp

```diff
@@ -0,0 +1,575 @@
+                {
+                    // Check current pixel for the current image mask
+                    if(_cellMasks[i].at<uchar>(y,x) == 255)
+                    {
+                        image::RGBAfColor& px = img(y + _bbox.min.y, x + _bbox.min.x);
+                        _colorData.at< cv::Vec3d >(i,0) += cv::Vec3d(px.r(), px.g(), px.b());
+                        ++pixelsCount.at<int>(i);
+                    }
+                }
+            }
+        }
+
+        for(int i = 0; i < _colorData.rows; ++i)
+            _colorData.at<cv::Vec3d>(i, 0) /= pixelsCount.at<int>(i); // average value
+    }
+
+    bpt::ptree ptree() const
+    {
+        bpt::ptree pt, ptColors, ptPositions, ptTransform;
+
+        pt.put("bodySerialNumber", _imgOpt.bodySerialNumber);
+        pt.put("lensSerialNumber", _imgOpt.lensSerialNumber);
+        pt.put("imagePath", _imgOpt.imgFsPath.string());
+        pt.put("viewId", _imgOpt.viewId);
+
+        // Serialize checker.positions
+        for (const auto& point : _cchecker->getBox())
+        {
+            bpt::ptree ptPoint;
+            ptPoint.put("x", point.x);
+            ptPoint.put("y", point.y);
+            ptPositions.push_back( std::make_pair("", ptPoint) );
+        }
+        pt.add_child("positions", ptPositions);
+
+        // Serialize checker.transform
+        for (int i = 0; i < _transformMat.rows; ++i)
+        {
+            bpt::ptree row;
+            for(int j = 0; j < _transformMat.cols; ++j)
+            {
+                bpt::ptree cell;
+                cell.put_value(_transformMat.at<double>(i, j));
+                row.push_back(std::make_pair("", cell));
+            }
+            ptTransform.push_back( std::make_pair("", row) );
+        }
+        pt.add_child("transform", ptTransform);
+
+        // Serialize colors
+        for (int i = 0; i < _colorData.rows; ++i)
+        {
+            bpt::ptree ptColor;
+            ptColor.put("r", _colorData.at<cv::Vec3d>(i,0)[0]);
+            ptColor.put("g", _colorData.at<cv::Vec3d>(i,0)[1]);
+            ptColor.put("b", _colorData.at<cv::Vec3d>(i,0)[2]);
+            ptColors.push_back( std::make_pair("", ptColor) );
+        }
+        pt.add_child("colors", ptColors);
+
+        return pt;
+    }
+};
+
+
+void detectColorChecker(
+    std::vector<MacbethCCheckerQuad> &detectedCCheckers,
+    ImageOptions& imgOpt,
+    CCheckerDetectionSettings &settings)
+{
+    const int nc = 2; // Max number of charts in an image
+    const std::string outputFolder = fs::path(settings.outputData).parent_path().string() + "/";
+    const std::string imgSrcPath = imgOpt.imgFsPath.string();
+    const std::string imgSrcStem = imgOpt.imgFsPath.stem().string();
+    const std::string imgDestStem = imgSrcStem;
+
+    // Load image
+    image::Image<image::RGBAfColor> img;
+    image::readImage(imgSrcPath, img, imgOpt.readOptions);
+    cv::Mat imgBGR = image::imageRGBAToCvMatBGR(img, CV_8UC3);
```

Comment:
To convert to 8 bits the image needs to be in sRGB in all case (so it cannot rely on imgOpt.readOptions input parameter).
To read the color values in floating point to compute the color transformation, the image needs to be in the original file format colorspace or in linear (just need to use the same convention where we apply it).